contracts/src/link3.rs

```
@@ -92,6 +92,34 @@ impl Link3 {
     self.links.clone()
   }
 
+  pub fn update(
+    &mut self,
+    title: String,
+    description: String,
+    image_uri: Option<String>,
+  ) -> &Link3 {
+    if env::signer_account_id() != self.owner_account_id {
+      panic!("Only the owner can update the contract.");
+    }
+
+    if self.title != title {
+      self.validate_title(&title);
```

Not sure I like these type of `validate_` methods.

I would prefer to have it return a boolean rather than silently `panic!`

---

This way we don't need to do an if statement on the parent

In any case we do not want to proceed if any condition fail so let's panic right away.